db-service/test/cqn4sql/expand.test.js

```diff
@@ -1092,6 +1092,93 @@ describe('Unfold expands on associations to special subselects', () => {
     }`
     expect(JSON.parse(JSON.stringify(cqn4sql(q, model)))).to.eql(expected)
   })
+
+  it('expand via subquery with path expressions', () => {
+    const q = cds.ql`SELECT from (SELECT from bookshop.Books as inner { author, ID } where author.name = 'King') as Outer {
+      ID,
+      author { name }
+    }`
+    const res = cqn4sql(q, model)
+    const expected = cds.ql`
+      SELECT from (
+        SELECT from bookshop.Books as inner
+          left join bookshop.Authors as author on author.ID = inner.author_ID {
+            inner.author_ID,
+            inner.ID
+          } where author.name = 'King'
+        ) as Outer
+      {
+        Outer.ID,
+        (
+          SELECT from bookshop.Authors as $a {
+            $a.name
+          }
+          where Outer.author_ID = $a.ID
+        ) as author
+      }`
+    expect(JSON.parse(JSON.stringify(res))).to.deep.equal(expected)
+  })
+  it('expand via subquery with path expressions nested', () => {
+    const q = cds.ql`SELECT from (SELECT from (SELECT from bookshop.Books as inner { author, ID } where author.name = 'King') as Mid { * }) as Outer {
```

Comment:
btw this revealed another issue, which is not subject to this PR. So I will ship a dedicated fix for #1127 once this is in main